### ISSUE TITLE
Cannot derive StateMachine for public struct

### ISSUE THREAD

**Issue by joonazan** [aper-dev/aper#22]
```
error[E0446]: private type `CharacterTransform` in public interface
 --> state/src/character.rs:4:10
  |
4 | #[derive(StateMachine, Clone, Debug, Serialize, Deserialize, PartialEq)]
  |          ^^^^^^^^^^^^
  |          |
  |          can't leak private type
  |          `CharacterTransform` declared as private
  |
  = note: this error originates in the derive macro `StateMachine` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by joonazan** [comment]
Solved by #24

**Comment by paulgb** [comment]
Excellent, thanks for confirming! I will merge now.
